src/test/ui/iterators/vec-on-unimplemented.stderr

```diff
@@ -0,0 +1,20 @@
+error[E0599]: `Vec<bool>` is not an iterator
+  --> $DIR/vec-on-unimplemented.rs:2:23
+   |
+LL |     vec![true, false].map(|v| !v).collect::<Vec<_>>();
+   |                       ^^^ `Vec<bool>` is not an iterator; try calling `.into_iter()` or `.iter()`
+   |
+  ::: $SRC_DIR/alloc/src/vec/mod.rs:LL:COL
+   |
+LL | pub struct Vec<T, #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global> {
```

Comment:
This span is strange...

Reply:
Are you talking about the reference to `Vec`'s declaration or to `map` in the code?

Reply:
The reference to the declaration. It's not present on nightly in playground or godbolt -- wonder if it's due to the use of `on_unimplemented` or just because of the UI test environment.

Reply:
IIRC correctly it was present even before I added my `on` clause, I may be wrong though. I think rustc considers alloc to be a local crate in test environment?

Reply:
Just checked - it does mention this even without this commit:
```
error[E0599]: `Vec<bool>` is not an iterator
   --> t.rs:11:23
    |
11  |     vec![true, false].map(|v| !v).collect::<Vec<_>>();
    |                       ^^^ `Vec<bool>` is not an iterator
    |
   ::: /workspaces/rust/library/alloc/src/vec/mod.rs:400:1
    |
400 | pub struct Vec<T, #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global> {
    | ------------------------------------------------------------------------------------------------ doesn't satisfy `Vec<bool>: Iterator`
    |
    = note: the following trait bounds were not satisfied:
            `Vec<bool>: Iterator`
            which is required by `&mut Vec<bool>: Iterator`
            `[bool]: Iterator`
            which is required by `&mut [bool]: Iterator`

error: aborting due to previous error
```